### PR TITLE
Updating documentation around module usage with Deno

### PR DESCRIPTION
**Pull request by thisisjofrank**
Adding docs on importing with a focus on JSR
Adding docs on publishing with a focus on JSR
Moved some of the smaller pages, for example the module metadata file into the importing page
Deleted old and outdated example, which is now covered in the above docs.

### CODE REVIEW

File: runtime/manual/basics/modules/publishing_modules.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+Title: Publishing Modules
```
Comment (ry) [inline runtime/manual/basics/modules/publishing_modules.md:2]: Use `oldUrl` here to add a redirect from `/runtime/manual/advanced/publishing/dnt/`

Untested, but something like this:
```
oldUrl:
  - /runtime/manual/advanced/publishing/dnt/
```

File: runtime/manual/basics/modules/remote_modules.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
+# Using Remote Modules with Deno
```
Comment (ry) [inline runtime/manual/basics/modules/remote_modules.md:1]: "Remote Modules" is not the most widely understandable term - consider changing the title of this section.

Maybe "Using third party modules" or "Using external code" or "Using libraries"

File: runtime/manual/basics/modules/index.md [hunk 3]
```diff
@@ -94,63 +70,3 @@ All functions, classes, constants and variables which need to be accessible
 inside external modules must be exported. Either by prepending them with the
 `export` keyword or including them in an export statement at the bottom of the
 file.
-
-## FAQ
```
Comment (ry) [inline runtime/manual/basics/modules/index.md:98]: I am also of the opinion that FAQs are evidence of bad documentation. Glad this is being removed : )

File: runtime/manual/basics/modules/remote_modules.md (hunk 2 above)
Comment (thisisjofrank) [reply runtime/manual/basics/modules/remote_modules.md:1]: Yeah, thinking about it, I agree, this is going to be the primary usecase. I'll move it into index.

File: runtime/manual/basics/modules/publishing_modules.md (hunk 1 above)
Comment (lucacasonato) [reply runtime/manual/basics/modules/publishing_modules.md:2]:
```suggestion
title: Publishing Modules
```

File: runtime/manual/basics/modules/index.md [hunk 4]
```diff
@@ -95,62 +71,168 @@ inside external modules must be exported. Either by prepending them with the
...
+  }
 }
 ```
 
-before running your project.
+Supporting URL imports enables us to support the following JavaScript CDNs, as
+they provide URL access to JavaScript modules:
+
+- [deno.land/x](https://deno.land/x)
+- [esm.h](https://esm.sh)
```
Comment (lucacasonato) [inline runtime/manual/basics/modules/index.md:215]:
```suggestion
- [esm.sh](https://esm.sh)
```

File: runtime/manual/basics/modules/publishing_modules.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+If you're configuring a CI pipeline to publish your module, you can use the
+`--token` flag to authenticate with JSR but it's important to ensure that your
+token is kept secure.
+
+If you're publishing a package from GitHub Actions it is recommended to link
+your repository by following the instructions at
+[jsr.io/docs/publishing-packages#publishing-from-github-actions](https://jsr.io/docs/publishing-packages#publishing-from-github-actions).
+This will allow you to publish packages without needing to provide an auth token
+and instead uses OIDC authentication from GitHub itself.
```
Comment (lucacasonato) [inline runtime/manual/basics/modules/publishing_modules.md:74]: Move the GH actions thing first, then the `--token` flag, and explicitly discourage people to use it (they should prefer GitHub and interactive publishing).

**Pull request merged**